contracts/base/Derivative.sol

```diff
@@ -107,6 +110,39 @@ contract Derivative is ERC721, Ownable, Versioned {
     nullifiers[nullifier] = true;
   }
 
+  function tokenURI(uint256 tokenId)
+    public
+    view
+    virtual
+    override
+    returns (string memory)
+  {
+    _requireMinted(tokenId);
+
+    string memory baseURI = _baseURI();
+    string memory contractAddress = Strings.toHexString(
+      uint160(address(this)),
+      20
+    );
+    return
+      bytes(baseURI).length > 0
+        ? string(
+          abi.encodePacked(baseURI, contractAddress, "/", tokenId.toString())
+        )
+        : "";
+  }
+
+  function _baseURI() internal view virtual override returns (string memory) {
+    return
+      bytes(derivativeBaseURI).length > 0
+        ? derivativeBaseURI
+        : ILedger(ledgerContract).getBaseURI();
+  }
+
+  function setBaseURI(string memory _newBaseURI) external onlyOwner {
```

Comment:
There is no need in prefixing function argument names with `new`. `_baseURI` is enough to distinguish it from the `baseURI` instance variable for the contract